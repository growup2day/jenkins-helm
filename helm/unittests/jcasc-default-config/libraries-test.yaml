templates:
  - config-jcasc.yaml
tests:
  - it: should include pipeline libraries config in default config scripts when configured
    set:
      JCasC:
        defaultConfig: true
        pipelineLibraries:
          pipeline-lib1:
            repo: https://bitbucket.nz.service.anz/some/dummy-pipeline.git
            credentialsId: "anz-bitbucket-cred"
            includeInChangesets: false
          pipeline-lib2:
            repo: https://bitbucket.nz.service.anz/some/other-dummy-pipeline.git
            includeInChangesets: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-libraries.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-libraries.yaml]
  - it: should include blank pipeline libraries config in default config scripts when not configured
    set:
      JCasC:
        defaultConfig: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-libraries.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-libraries.yaml]
  - it:
      should not include pipeline libraries config in default config scripts when there is custom config script for the
      section
    set:
      JCasC:
        defaultConfig: true
        configScripts:
          custom-override: |
            unclassified:
              globalLibraries:
                libraries: []
    asserts:
      - isNull:
          path: data.[jcasc-default-config-libraries.yaml]
      - isNotNull:
          path: data.[custom-override.yaml]
