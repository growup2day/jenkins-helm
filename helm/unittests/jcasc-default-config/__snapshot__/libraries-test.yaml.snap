should include blank pipeline libraries config in default config scripts when not configured:
  1: |
    |-
      unclassified:
        # Configure System > Global Pipeline Libraries
        globalLibraries:
          libraries: []
should include pipeline libraries config in default config scripts when configured:
  1: |
    |-
      unclassified:
        # Configure System > Global Pipeline Libraries
        globalLibraries:
          libraries:
            - name: "pipeline-lib1"
              defaultVersion: "master"
              includeInChangesets: false
              retriever:
                modernSCM:
                  scm:
                    git:
                      credentialsId: "anz-bitbucket-cred"
                      remote: "https://bitbucket.nz.service.anz/some/dummy-pipeline.git"
                      traits:
                        - "gitBranchDiscovery"
            - name: "pipeline-lib2"
              defaultVersion: "master"
              includeInChangesets: true
              retriever:
                modernSCM:
                  scm:
                    git:
                      remote: "https://bitbucket.nz.service.anz/some/other-dummy-pipeline.git"
                      traits:
                        - "gitBranchDiscovery"
